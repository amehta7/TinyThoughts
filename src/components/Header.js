import React from 'react'
import AccountNav from './AccountNav'
import { Link } from 'react-router-dom'

const Header = () => {
  return (
    <div className='header'>
      <div className='app-title'>TinyThoughts</div>
      <div className='app-nav'>
        <ul>
          <li>
            <Link to='/'>Home</Link>
          </li>
          <li>
            <Link to='/add'>New Post</Link>
          </li>
        </ul>
      </div>

      <AccountNav />
    </div>
  )
}

export default Header
