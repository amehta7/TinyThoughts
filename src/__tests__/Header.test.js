import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Header from "../components/Header";

const mockStore = configureStore([thunk]);

describe("Header component", () => {
  it("Header component renders correctly when signed in", () => {
    const store = mockStore({
      users: {
        user: { id: 1, name: "Joe Mockery" }
      },
      posts: {
        categories: [
          { id: 1, title: "JavaScript" },
          { id: 2, title: "Python" }
        ]
      },
      errors: {
        error: false
      }
    });

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Header />
        </MemoryRouter>
      </Provider>
    );

    return expect(wrapper.find(Header)).toMatchSnapshot();
  });

  it("Header component renders correctly when signed out", () => {
    const store = mockStore({
      users: {},
      posts: {
        categories: [
          { id: 1, title: "JavaScript" },
          { id: 2, title: "Python" }
        ]
      },
      errors: {
        error: false
      }
    });

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Header />
        </MemoryRouter>
      </Provider>
    );

    return expect(wrapper.find(Header)).toMatchSnapshot();
  });
});
