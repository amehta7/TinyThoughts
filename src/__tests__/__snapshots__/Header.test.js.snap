// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header component renders correctly when signed in 1`] = `
<Header>
  <div
    className="header"
  >
    <div
      className="app-title"
    >
      TinyThoughts
    </div>
    <div
      className="app-nav"
    >
      <ul>
        <Link
          to="/"
        >
          <LinkAnchor
            href="/"
            navigate={[Function]}
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <li>
                Home
              </li>
            </a>
          </LinkAnchor>
        </Link>
        <Link
          to="/add"
        >
          <LinkAnchor
            href="/add"
            navigate={[Function]}
          >
            <a
              href="/add"
              onClick={[Function]}
            >
              <li>
                New Post
              </li>
            </a>
          </LinkAnchor>
        </Link>
      </ul>
    </div>
    <Memo()>
      <div
        className="account-nav"
      >
        <h4>
          Joe Mockery
        </h4>
        <ul>
          <li
            onClick={[Function]}
          >
            Signout
          </li>
        </ul>
      </div>
    </Memo()>
  </div>
</Header>
`;

exports[`Header component Header component renders correctly when signed out 1`] = `
<Header>
  <div
    className="header"
  >
    <div
      className="app-title"
    >
      TinyThoughts
    </div>
    <div
      className="app-nav"
    >
      <ul>
        <Link
          to="/"
        >
          <LinkAnchor
            href="/"
            navigate={[Function]}
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <li>
                Home
              </li>
            </a>
          </LinkAnchor>
        </Link>
        <Link
          to="/add"
        >
          <LinkAnchor
            href="/add"
            navigate={[Function]}
          >
            <a
              href="/add"
              onClick={[Function]}
            >
              <li>
                New Post
              </li>
            </a>
          </LinkAnchor>
        </Link>
      </ul>
    </div>
    <Memo()>
      <div
        className="account-nav"
      >
        <ul>
          <Link
            to="/signin"
          >
            <LinkAnchor
              href="/signin"
              navigate={[Function]}
            >
              <a
                href="/signin"
                onClick={[Function]}
              >
                <li>
                  Signin
                </li>
              </a>
            </LinkAnchor>
          </Link>
          <Link
            to="/signup"
          >
            <LinkAnchor
              href="/signup"
              navigate={[Function]}
            >
              <a
                href="/signup"
                onClick={[Function]}
              >
                <li>
                  Signup
                </li>
              </a>
            </LinkAnchor>
          </Link>
        </ul>
      </div>
    </Memo()>
  </div>
</Header>
`;
